### ISSUE TITLE
List all group folder on settings take a long time

### ISSUE THREAD

**Issue by Gransi** [nextcloud/groupfolders#130]
Hello,

When we open the group folder overview on settings, then we must wait up to 4 minutes until we can edit the group folders. On furhter investigation we found the reason. 

![nc_groupfolder](https://user-images.githubusercontent.com/33392356/41980638-42ca3366-7a27-11e8-80d4-055a163a30c2.JPG)

The function getAllFolders() in FolderManager.php makes an join to the filecache table on mariadb and our table has up to 15 million rows (8.5GB), therefore needs the query a long time.

Temporaly I have removed the "leftjoin" and "size".

```php
	public function getAllFolders() {
		$applicableMap = $this->getAllApplicable();

		$query = $this->connection->getQueryBuilder();

		$folderPath = $query->func()->concat($query->createNamedParameter('__groupfolders/'), 'folder_id');

		$query->select('folder_id', 'mount_point', 'quota')
			->from('group_folders', 'f');			

		$rows = $query->execute()->fetchAll();

		$folderMap = [];
		foreach ($rows as $row) {
			$id = $row['folder_id'];
			$folderMap[$id] = [
				'id' => $id,
				'mount_point' => $row['mount_point'],
				'groups' => isset($applicableMap[$id]) ? $applicableMap[$id] : [],
				'quota' => $row['quota'],
				'size' => $row['size'] ? $row['size'] : 0
			];
		}

		return $folderMap;
	}
```

**Comment by icewind1991** [comment]
Should be fixed with https://github.com/nextcloud/groupfolders/pull/131

**Comment by Gransi** [comment]
With the new fix I found a new bug. I can't make comments on folder on files.

The old version works.

**Comment by icewind1991** [comment]
Comments should be fixed with e06ecaf

**Comment by Gransi** [comment]
Thank you.
Now overview, tagging, comments are working
